File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/db/DBReceptores.java
package com.valleapp.vallecom.db;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

/**
 * Created by valle on 13/10/14.
 */
public class DBReceptores extends DBBase {

    public DBReceptores(Context context) {
        super(context, "receptores");
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE IF NOT EXISTS receptores (ID INTEGER PRIMARY KEY, nombre TEXT)");
    }

    @SuppressLint("Range")
    @Override
    protected JSONObject cursorToJSON(Cursor res) {
        JSONObject receptor = new JSONObject();
        try {
            receptor.put("nombre", res.getString(res.getColumnIndex("nombre")));
            receptor.put("ID", res.getString(res.getColumnIndex("ID")));
        }catch (Exception e){
            e.printStackTrace();
        }

        return receptor;
    }

    @Override
    protected ContentValues caragarValues(JSONObject o) {
        ContentValues v = new ContentValues();
        try {
            v.put("nombre", o.getString("nombre"));
            v.put("ID", o.getString("ID"));
        }catch (Exception e){
            e.printStackTrace();
        }
        return  v;
    }


    public ArrayList<JSONObject> getAll() {
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor res =  db.rawQuery( "select * from receptores WHERE nombre NOT LIKE '%Nulo%' ", null );
        res.moveToFirst();
        ArrayList<JSONObject>  ls = new ArrayList<>();
        res.moveToFirst();
        while(!res.isAfterLast()){
            ls.add(cursorToJSON(res));
            res.moveToNext();
        }
        return ls;
    }


}

File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/db/DBReceptores.kt
package com.valleapp.valletpvlib.db

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import org.json.JSONObject

class DBReceptores(context: Context) : DBBase(context, "receptores") {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE IF NOT EXISTS receptores (ID INTEGER PRIMARY KEY, nombre TEXT, nomimp TEXT)")
    }

    override fun cursorToJSON(res: Cursor): JSONObject {
        val receptor = JSONObject()
        try {
            receptor.put("nombre", res.getString(with(res) { getColumnIndex("nombre") }))
            receptor.put("nomimp", res.getString(with(res) { getColumnIndex("nomimp") }))
            receptor.put("ID", res.getString(with(res) { getColumnIndex("ID") }))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return receptor
    }

    override fun cargarValues(o: JSONObject): ContentValues {
        val v = ContentValues()
        try {
            v.put("nombre", o.getString("nombre"))
            v.put("nomimp", o.getString("nomimp"))
            v.put("ID", o.getString("id"))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return v
    }

    fun getAll(): ArrayList<JSONObject> {
        val db = this.readableDatabase
        val res = db.rawQuery("select * from receptores WHERE nomimp!='Nulo' and nomimp!='None' and nomimp!='' and nombre!='Ticket' " , null)
        val ls = ArrayList<JSONObject>()
        try {
            res.moveToFirst()
            while (!res.isAfterLast) {
                ls.add(cursorToJSON(res))
                res.moveToNext()
            }
        } finally {
            res.close()
        }
        return ls
    }

}